crates/solver/src/driver.rs

```diff
@@ -609,6 +610,50 @@ impl Driver {
 
         rated_settlements.sort_by(|a, b| a.1.objective_value().cmp(&b.1.objective_value()));
         print_settlements(&rated_settlements, &self.fee_objective_scaling_factor);
+
+        // Report solver competition data to the api.
+        let mut solver_competition_response = SolverCompetitionResponse {
+            gas_price: gas_price.effective_gas_price(),
+            liquidity_collected_block: current_block_during_liquidity_fetch,
+            // TODO: we don't have access to this and there is no guarantee there is one such block
+            competition_simulation_block: 0,
```

Comment:
Regarding this TODO and the one below - do you think it makes sense to return this along with the simulations that we already do?

Specifically we can add a `eth_getBlockNumber(latest)` to the request batch and return the computed calldata along with the simulation result that computes the gas used.

You mentioned below that some refactoring to make this happen, I'm not sure if this is what you had in mind.

Reply:
For both the todos we need a little bit of refactoring yeah. The reason here it doesn't completely work is that we split up over multiple batches if there are too many simulations. These batches could happen in different block numbers. (Really we don't even have a guarantee that a batch always happes in the same block either.)

Reply:
> Really we don't even have a guarantee that a batch always happes in the same block either.

Oh yeah, that's a good point. It may be nice as an indicative value anyway even if its not accurate.